### PR TITLE
Remove indirection for accessing str / bytes.

### PR DESCRIPTION
**Pull request by okin**
# Pull Request Checklist

- [x] I’ve read the [guidelines for contributing](https://github.com/getnikola/nikola/blob/master/CONTRIBUTING.rst).
- [x] I updated AUTHORS.txt and CHANGES.txt (if the change is non-trivial) and documentation (if applicable).
- [x] I tested my changes.

### Description

With Nikola being Python 3 only this isn't necessary anymore.

This also makes attempting #3327 easier because it reduces some smaller dependency to `nikola.utils`.

### CODE REVIEW

File: nikola/utils.py [hunk 1]
```diff
@@ -104,11 +104,6 @@
 # Are you looking for 'generic_rss_renderer'?
 # It's defined in nikola.nikola.Nikola (the site object).
 
-# Aliases, previously for Python 2/3 compatibility.
```
Comment (Kwpolska) [inline nikola/utils.py]: I don’t think we can remove this, plugins could still use those.

Comment (okin) [reply nikola/utils.py]: I checked for those in getnikola/plugins.
Maybe this is a change that should be held off until a larger release.

Comment (Kwpolska) [reply nikola/utils.py]: We can change everything now, add a standard `# TODO: remove in v9` deprecation notice to the codebase, and that will be taken care of in the future.

Comment (Kwpolska) [reply nikola/utils.py]: I’ll handle this from here.

**Pull request merged**